packages/oslo-generator-json-webuniversum/lib/JsonWebuniversumGenerationService.ts

```diff
@@ -0,0 +1,197 @@
+    }
+
+    const fetchProperty = (fetchFunction: (subjectId: RDF.NamedNode, store: QuadStore, language: string) => RDF.Literal | undefined, subject: RDF.NamedNode) =>
+      fetchFunction(subject, this.store, this.configuration.language)?.value;
+
+    const vocabularyLabel: string | undefined = fetchProperty(getVocabularyLabel, entity);
+    const applicationProfileLabel: string | undefined = fetchProperty(getApplicationProfileLabel, entity);
+    const vocabularyDefinition: string | undefined = fetchProperty(getVocabularyDefinition, entity);
+    const applicationProfileDefinition: string | undefined = fetchProperty(getApplicationProfileDefinition, entity);
+    const vocabularyUsageNote: string | undefined = fetchProperty(getVocabularyUsageNote, entity);
+    const applicationProfileUsageNote: string | undefined = fetchProperty(getApplicationProfileUsageNote, entity);
+
+    const parentsIds: RDF.NamedNode[] = includeProperties ? this.store.getParentsOfClass(entity) : this.store.getParentOfProperty(entity) !== undefined ? [this.store.getParentOfProperty(entity)!] : [];
+    const parentObjects: Pick<WebuniversumObject, 'id' | 'vocabularyLabel' | 'applicationProfileLabel'>[] = parentsIds.map(parentId => this.createParentObject(parentId));
+
+    const scope: string | undefined = this.store.getScope(entity)?.value;
+
+    const entityData: WebuniversumObject = {
+      id: assignedURI.value,
+      ...vocabularyLabel && {
+        vocabularyLabel: { [this.configuration.language]: vocabularyLabel },
+      },
+      ...applicationProfileLabel && {
+        applicationProfileLabel: { [this.configuration.language]: applicationProfileLabel },
+      },
+      ...vocabularyDefinition && {
+        vocabularyDefinition: { [this.configuration.language]: vocabularyDefinition },
+      },
+      ...applicationProfileDefinition && {
+        applicationProfileDefinition: { [this.configuration.language]: applicationProfileDefinition },
+      },
+      ...vocabularyUsageNote && {
+        vocabularyUsageNote: { [this.configuration.language]: vocabularyUsageNote },
+      },
+      ...applicationProfileUsageNote && {
+        applicationProfileUsageNote: { [this.configuration.language]: applicationProfileUsageNote },
+      },
+      ...parentObjects.length > 0 && {
+        parents: parentObjects
+      },
+      ...scope && {
+        scope: scope,
+      }
+    };
+
+    if (includeProperties) {
+      const jobs: Promise<WebuniversumProperty>[] = this.store.findSubjects(ns.rdfs('domain'), entity)
+        .map<Promise<WebuniversumProperty>>(async (property: RDF.Term) => {
+          return <Promise<WebuniversumProperty>>this.generateEntityData(<RDF.NamedNode>property, false)
+            .then((propertyObject) => this.addPropertySpecificInformation(<RDF.NamedNode>property, <WebuniversumProperty>propertyObject, assignedURI));
+        });
+
+      const properties: WebuniversumProperty[] = await Promise.all(jobs)
+
+      if (properties.length > 0) {
```

Comment:
The extra length check on `> 0` should not be needed here. `properties.length` does the same